File: data_structure/Queue/python/queue.py
import sys, os
sys.path.append(os.path.dirname(os.path.abspath(os.path.dirname(__file__))))

from linked_list import doubly_linked_list
DoublyLinkedList = doubly_linked_list.DoublyLinkedList


class Queue:
    def __init__(self):
        self.list = DoublyLinkedList()

    def enqueue(self, data):
        self.list.insert_at(0, data)

    def dequeue(self):
        try:
            return self.list.delete_last()
        except IndexError as e:
            return None

    def front(self):
        return self.list.tail

    def is_empty(self):
        return self.list.count == 0

File: data_structure/Queue/java/Queue.java
package data_structure.queue;

import data_structure.linked_list.doubly_linked_list.DoublyLinkedList;
import data_structure.linked_list.doubly_linked_list.Node;

public class Queue <T> {
    private DoublyLinkedList<T> list = new DoublyLinkedList<>();

    /**
     * 데이터 삽입
     */
    public void enqueue(T data) {
        list.insertAt(0, data);
    }

    /**
     * 데이터 삭제
     */
    public Node<T> dequeue() {
        try {
            return list.deleteLast();
        } catch (Exception e) {
            return null;
        }
    }

    /**
     * 데이터 확인
     */
    public Node front() {
        return list.getTail();
    }

    /**
     * 리스트가 비었는지 점검
     */
    public boolean isEmpty() {
        return list.getCount() == 0;
    }
}
